posix-mapper/src/intTest/java/org/opencadc/posix/mapper/UserManagementIntTest.java

```diff
@@ -0,0 +1,194 @@
+        userSubject = new Subject();
+        userSubject.getPublicCredentials().add(
+                new AuthorizationToken("bearer", bearerToken, List.of(NetUtil.getDomainName(userMapperURL))));
+        log.debug("userSubject: " + userSubject);
+    }
+
+    final String randomUsername() {
+        return RandomStringUtils.randomAlphabetic(4, 12);
+    }
+
+    @Test
+    public void testNotAuthenticated() throws Exception {
+        try {
+            createUser(randomUsername());
+            Assert.fail("Should throw NotAuthenticatedException");
+        } catch (NotAuthenticatedException notAuthenticatedException) {
+            // Good
+        } catch (Throwable throwable) {
+            throw new Exception(throwable.getMessage(), throwable);
+        }
+    }
+
+    @Test
+    public void testUserAdd() throws Exception {
+        Subject.doAs(userSubject, (PrivilegedExceptionAction<Void>) () -> {
+            final String username = randomUsername();
+            try {
+                final int uid = createUser(username);
+                final ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+                getUsers(byteArrayOutputStream, UserManagementIntTest.TEXT_PLAIN_CONTENT_PLAIN);
+                String output = byteArrayOutputStream.toString();
+                Assert.assertTrue("Wrong output",
+                                  output.contains(String.format("%s:x:%d:%d:::", username, uid, uid)));
+                getUsers(byteArrayOutputStream, UserManagementIntTest.TSV_CONTENT_PLAIN);
+                output = byteArrayOutputStream.toString();
+                Assert.assertTrue("Wrong TSV output",
+                                  output.contains(String.format("%s\t%d", username, uid)));
+            } catch (Throwable throwable) {
+                throw new Exception(throwable.getMessage(), throwable);
+            }
+
+            return null;
+        });
+    }
+
+    private int createUser(final String username) throws Throwable {
+        final Map<String, Object> payload = new HashMap<>();
+
+        payload.put("username", username);
+
+        final ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+        final HttpPost httpPost = new HttpPost(userMapperURL, payload, byteArrayOutputStream);
+        httpPost.run();
+
+        if (httpPost.getThrowable() != null) {
+            throw httpPost.getThrowable();
+        }
+
+        return Integer.parseInt(byteArrayOutputStream.toString());
+    }
+
+    private void getUsers(final OutputStream outputStream, final String contentType) throws Throwable {
+        final InputStreamWrapper inputStreamWrapper = inputStream -> outputStream.write(inputStream.readAllBytes());
+        final HttpGet httpGet = new HttpGet(userMapperURL, inputStreamWrapper);
+        httpGet.setRequestProperty("accept", contentType);
+        httpGet.run();
```

Comment:
HttpTransfer(s) support a new more direct mode:

instead of calling `run()` you can call `prepare()` which does everything except read the stream and throws a variety of exceptions; then you don't haveto check `getThrowable()`. If `prepare()` returns (success) then you can call `getInputStream()` and check it or (like here) just close it).